lib/storage/bucket.js

```diff
@@ -198,6 +198,10 @@ Bucket.prototype.delete = function(callback) {
  * var file = bucket.file('my-existing-file.png');
  */
 Bucket.prototype.file = function(name) {
+  if (!name) {
```

Comment:
He's just moving the error handling to the method that can actually cause it. A File is only directly instantiated by a user through this method, so it makes sense to me to throw the error here. I'm fine with it staying where it is as well. I thought that's what we were letting him decide. Checking for a bucket isn't necessary, since there will always be a bucket in order to create a file.